pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -264,13 +275,28 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListeners(env model.Env
 		listenerType = plugin.ModelProtocolToListenerType(protocol)
 		switch listenerType {
 		case plugin.ListenerTypeHTTP:
-			listenerOpts.filterChainOpts = []*filterChainOpts{{
-				httpOpts: &httpListenerOpts{
-					routeConfig:      configgen.buildSidecarInboundHTTPRouteConfig(env, node, instance),
-					rds:              "",
-					useRemoteAddress: false,
-					direction:        http_conn.INGRESS,
-				}},
+			httpOpts := &httpListenerOpts{
+				routeConfig:      configgen.buildSidecarInboundHTTPRouteConfig(env, node, instance),
+				rds:              "",
+				useRemoteAddress: false,
+				direction:        http_conn.INGRESS,
+			}
+			if listenerOpts.tlsMultiplexed {
+				listenerOpts.filterChainOpts = []*filterChainOpts{
+					{
+						httpOpts:          httpOpts,
+						transportProtocol: authn.EnvoyRawBufferMatch,
+					}, {
+						httpOpts:          httpOpts,
```

Comment:
@ostromart I am thinking of something like this:
OnFilterChainOpts callback.. We construct the default one as usual (with httpopts). Auth plugin could overwrite it and make it into two filterchainopts, and we can have a filterChainOpts.pluginOpts field where the plugins could store arbitrary data.

Then in buildListener, we again call plugin.AddListenerFilters with the filterChainOpts as argument. Auth plugin could once again install the TLS inspector.